launcher/minecraft/launch/LauncherPartLaunch.cpp

```diff
@@ -159,6 +161,113 @@ void LauncherPartLaunch::executeTask()
+        bwrapArgs << "--unshare-all";
+        bwrapArgs << "--share-net";
+        bwrapArgs << "--die-with-parent";
+        bwrapArgs << "--unsetenv" << "DBUS_SESSION_BUS_ADDRESS";
+
+        // default binds
+        bwrapArgs << "--dev" << "/dev";
+        bwrapArgs << "--dev-bind-try" << "/dev/dri" << "/dev/dri";
+        bwrapArgs << "--proc" << "/proc";
+        bwrapArgs << "--setenv" << "XDG_RUNTIME_DIR" << sandboxedRuntimeDir;
+
+        for (auto path : systemBinds) {
+            bwrapArgs << "--ro-bind-try" << path << path;
+        }
+
+        // desktop integration
+        bwrapArgs << "--ro-bind-try" << QString("%1/pulse").arg(actualRuntimeDir) << QString("%1/pulse").arg(sandboxedRuntimeDir);
+        bwrapArgs << "--ro-bind-try" << QString("%1/pipewire-0").arg(actualRuntimeDir) << QString("%1/pipewire-0").arg(sandboxedRuntimeDir);
+        {
+            auto display = qEnvironmentVariable("DISPLAY");
+            auto xAuthPath = qEnvironmentVariable("XAUTHORITY",
+                    FS::PathCombine(QStandardPaths::writableLocation(QStandardPaths::HomeLocation), ".Xauthority"));
+            auto wlDisplay = qEnvironmentVariable("WAYLAND_DISPLAY");
+
+            if (display.startsWith(':')) {
+                auto x11Socket = QString("/tmp/.X11-unix/X%1").arg(display.mid(1));
+                bwrapArgs << "--ro-bind-try" << x11Socket << x11Socket;
+            }
+
+            bwrapArgs << "--ro-bind-try" << xAuthPath << xAuthPath;
+
+            if (wlDisplay.startsWith('/'))
+                bwrapArgs << "--ro-bind-try" << wlDisplay << wlDisplay;
+            else
+                bwrapArgs << "--ro-bind-try" << FS::PathCombine(actualRuntimeDir, wlDisplay) << FS::PathCombine(sandboxedRuntimeDir, wlDisplay);
+        }
+
+        if (minecraftInstance->settings()->get("EnableMangoHud").toBool() && APPLICATION->capabilities() & Application::SupportsMangoHud) {
+            auto mangoHudConfigPath = qEnvironmentVariable("MANGOHUD_CONFIGFILE",
+                    FS::PathCombine(QStandardPaths::writableLocation(QStandardPaths::GenericConfigLocation), "MangoHud")); 
+            bwrapArgs << "--ro-bind-try" << mangoHudConfigPath << mangoHudConfigPath;
+        }
+
+        // launcher
+        {
+            QString instPath = QDir::toNativeSeparators(QDir(minecraftInstance->instanceRoot()).absolutePath());
+            QString assetsPath = QDir::toNativeSeparators(QDir("assets").absolutePath());
+            QString librariesPath = QDir::toNativeSeparators(QDir("libraries").absolutePath());
+
+            bwrapArgs << "--bind" << instPath << instPath;
+            bwrapArgs << "--ro-bind" << assetsPath << assetsPath;
+            bwrapArgs << "--ro-bind" << librariesPath << librariesPath;
+
+            // also add NewLaunch.jar just to be safe. This is probably covered by /usr or LINUX_BWRAP_EXTRA_ARGS though
+            bwrapArgs << "--ro-bind" << jarPath << jarPath;
+        }
+
```

Comment:
```suggestion

bwrapArgs << "--setenv" << "PRISM_INSTANCE_SANDBOXED" << "1";
```

Reply:
This would allow other programs to determine that they are running inside of a sandbox and change their behaviour accordingly

Reply:
maybe we could make this a more generic variable for other launchers that are also implementing bwrap to follow. it'd give mod devs a bit less work :)

Reply:
> maybe we could make this a more generic variable for other launchers that are also implementing bwrap to follow. it'd give mod devs a bit less work :)

Done!

Reply:
I think having `MINECRAFT_SANDBOX_VARIANT` with the value describing the sandbox is more valuable, as different implementations might have different quirks.

With Prism Launcher it will now be `MINECRAFT_SANDBOX_VARIANT=prismlauncher-bwrap`